onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorSlice.cpp

```diff
@@ -6,16 +6,26 @@
 namespace Dml
 {
 
-class DmlOperatorSlice : public DmlOperator, public SliceHelper
+template <uint32_t opsetVersion>
+class DmlOperatorSlice : public DmlOperator, public SliceHelperBase
 {
 public:
     DmlOperatorSlice(const MLOperatorKernelCreationContext& kernelInfo)
     :   DmlOperator(kernelInfo),
-        SliceHelper(kernelInfo, kernelInfo.GetTensorShapeDescription())
+        SliceHelperBase(kernelInfo, kernelInfo.GetTensorShapeDescription(), opsetVersion)
     {
-        ML_CHECK_VALID_ARGUMENT(kernelInfo.GetInputCount() == 1);
+        uint32_t minInputCount = (opsetVersion < 10) ? 1 : 3;
+        ML_CHECK_VALID_ARGUMENT(kernelInfo.GetInputCount() >= minInputCount);
         ML_CHECK_VALID_ARGUMENT(kernelInfo.GetOutputCount() == 1);
-        DmlOperator::Initialize(kernelInfo);
+
+        // TODO (23108599): Slice V10 introduces an optional "Steps" input which the kernel does not yet support.
+        if (kernelInfo.GetInputCount() > 4)
+        {
+            THROW_HR(E_NOTIMPL);
```

Comment:
[minor nit] You used `THROW_HR_IF(E_NOTIMPL, ceilMode != 0);` elsewhere. Could make this:

`THROW_HR_IF(E_NOTIMPL, kernelInfo.GetInputCount() > 4);`